### PR TITLE
Only accept trusted validators

### PR DESCRIPTION
**Pull request by callistonianembrace**
## Depends

- [ ] #182

## Problem

PR #182 adds support for adding and removing validators, but doesn't check if membership changes are trusted. 

## Solution

This PR builds on top of #182 and #184 to only accept trusted validators in Add_validators operation.

### CODE REVIEW

**Comment by EduardoRFS** [comment]
The commit / PR order needs to be different, otherwise if we need to revert this, it wouldn't be possible, and this is a critical change.

**Comment by callistonianembrace** [comment]
Understood. I dont' mind closing the other PRs and squash the main one into one commit.

**Review by anchpop — APPROVED** [review]

**Comment by callistonianembrace** [comment]
@EduardoRFS Commits re-ordered

**Comment by callistonianembrace** [comment]
182 and this PR need to be merged together into one PR. Let me know whats to be done next

File: node/building_blocks.re [hunk 1]
```diff
@@ -58,6 +58,22 @@ let is_current_producer = (state, ~key) => {
...
 // TODO: check if block must have published a new snapshot
 let is_signable = (state, block) => {
+  open Protocol.Operation.Side_chain;
+  let trusted_validators = state.Node.trusted_validators;
+  let rec contains_only_trusted_add_validator_op = result_so_far =>
+    fun
+    | [] => result_so_far
+    | [h, ...t] =>
+      switch (h.kind) {
+      | Add_validator(validator) =>
```
Comment (EduardoRFS) [inline node/building_blocks.re:68]: Remove_validator should also be validated, otherwise someone could remove all the validators of the chain

**Review by EduardoRFS — CHANGES_REQUESTED** [review]

File: node/building_blocks.re (hunk 1 above)
Comment (callistonianembrace) [reply node/building_blocks.re:68]: Addressed in https://github.com/marigold-dev/sidechain/pull/182

**Comment by callistonianembrace** [comment]
Superseded by https://github.com/marigold-dev/sidechain/pull/182

**Pull request closed without merging**